### PR TITLE
[ONNX] Update test requirements

### PR DESCRIPTION
**Pull request by kshpv**

### CODE REVIEW

**Review by MaximProshin — APPROVED** [review]

File: tests/onnx/requirements.txt [hunk 1]
```diff
@@ -2,11 +2,9 @@ pytest-mock>=3.3.1
 pytest-cov
 pytest-dependency>=0.5.1
 pytest-ordering>=0.6
-torch~=1.13.0
-torchvision~=0.14.0
+torch==2.1.0
+torchvision==0.16.0
```
Comment (KodiaqQ) [inline tests/onnx/requirements.txt:6]: What is the reason for these changes and the PR in general?

File: tests/onnx/data/reference_graphs/original_nncf_graph/densenet121.dot [hunk 2]
```diff
@@ -42,343 +42,337 @@ strict digraph  {
...
-"371 /features/denseblock4/denselayer16/conv2/Conv" [id=371, type=Conv];
-"372 /features/denseblock4/Concat" [id=372, type=Concat];
-"373 /features/norm5/BatchNormalization" [id=373, type=BatchNormalization];
-"374 /Relu" [id=374, type=Relu];
-"375 /GlobalAveragePool" [id=375, type=GlobalAveragePool];
-"376 /Flatten" [id=376, type=Flatten];
-"377 /classifier/Gemm" [id=377, type=Gemm];
-"378 nncf_model_input_0" [id=378, type=nncf_model_input];
-"379 nncf_model_output_0" [id=379, type=nncf_model_output];
+"43 /features/transition1/pool/AveragePool" [id=43, type=AveragePool];
```
Comment (KodiaqQ) [inline tests/onnx/data/reference_graphs/original_nncf_graph/densenet121.dot:45]: Is there any understanding of the consequences of these architectural changes?
Do the patterns remain the same? What about the Quantizer's placement? What about the accuracy?

File: tests/onnx/requirements.txt (hunk 1 above)
Comment (kshpv) [reply tests/onnx/requirements.txt:6]: To validate the latest package versions

File: tests/onnx/data/reference_graphs/original_nncf_graph/densenet121.dot (hunk 2 above)
Comment (kshpv) [reply tests/onnx/data/reference_graphs/original_nncf_graph/densenet121.dot:45]: I took a look at the new FQ placements and it looks good! We didn't validate the metrics in these tests

Comment (KodiaqQ) [reply tests/onnx/data/reference_graphs/original_nncf_graph/densenet121.dot:45]: But we have E2E for ONNX. Have you looked at it? Or they contains a different Torch version as a requirement?

Comment (kshpv) [reply tests/onnx/data/reference_graphs/original_nncf_graph/densenet121.dot:45]: I ran a job and waiting for the results - job num 546.
But for E2E we do not use torch and torchvision, but I run to be sure that there is no problem with removing `Jinga2`

**Review by KodiaqQ — APPROVED** [review]
